src/modules/fancyzones/lib/util.cpp

```diff
@@ -63,7 +65,95 @@ namespace FancyZonesUtils
             return defaultDeviceId;
         }
     }
+    
+    std::optional<FancyZonesDataTypes::DeviceIdData> ParseDeviceId(const std::wstring& str)
```

Comment:
Is this complete new code? If yes, we need unit tests for it to cover all known cases.